crawler/priority_samples_process.py

```diff
@@ -0,0 +1,267 @@
+        # remove the lookup document
+        {"$project": {"related_samples": 0}}
+        # result = { mongo_sample_id=1234, must_sequence=true, uuid: 132 }
+    ]
+
+    value = priority_samples_collection.aggregate(IMPORTANT_UNPROCESSED_SAMPLES_MONGO_QUERY)
+    return list(value)
+
+
+def merge_priority_samples_into_docs_to_insert(priority_samples: List[Any], docs_to_insert: list) -> None:
+    """
+    Updates the sample records with must_sequence and preferentially_sequence values
+
+    for each successful add sample, merge into docs_to_insert_mlwh
+    with must_sequence and preferentially_sequence values
+
+    Arguments:
+        priority_samples  - priority samples to update docs_to_insert with
+        docs_to_insert {List[ModifiedRow]} -- the sample records to update
+    """
+
+    def extract_root_sample_id(sample: SampleDoc) -> SampleDocValue:
+        return sample[FIELD_ROOT_SAMPLE_ID]
+
+    priority_root_sample_ids = list(map(extract_root_sample_id, priority_samples))
+
+    for doc in docs_to_insert:
+        root_sample_id = doc[FIELD_ROOT_SAMPLE_ID]
+        if root_sample_id in priority_root_sample_ids:
+            priority_sample = list(filter(lambda x: x[FIELD_ROOT_SAMPLE_ID] == root_sample_id, priority_samples))[0]
+            doc[FIELD_MUST_SEQUENCE] = priority_sample[FIELD_MUST_SEQUENCE]
+            doc[FIELD_PREFERENTIALLY_SEQUENCE] = priority_sample[FIELD_PREFERENTIALLY_SEQUENCE]
+
+
+# TODO: refactor duplicated function
+def update_unprocessed_priority_samples_to_processed(db: Database, sample_ids: list) -> None:
+    """
+    Description
+    use stored identifiers to update priority_samples table to processed true
+    Arguments:
+        x {Type} -- description
+    """
+    priority_samples_collection = get_mongo_collection(db, COLLECTION_PRIORITY_SAMPLES)
+    for sample_id in sample_ids:
+        priority_samples_collection.update({FIELD_SAMPLE_ID: sample_id}, {"$set": {FIELD_PROCESSED: True}})
+    logger.info("Mongo update of processed for priority samples successful")
+
+
+# TODO: refactor duplicated function
+def update_priority_samples_into_mlwh(samples: List[Any], config: Config) -> bool:
+    """Insert sample records into the MLWH database from the parsed file information, including the corresponding
+    mongodb _id
+    Create all samples in MLWH with samples including must_seq/ pre_seq
+
+    Arguments:
+        samples {List[ModifiedRow]} -- List of filtered sample information extracted from CSV files.
+
+    Returns:
+        {bool} -- True if the insert was successful; otherwise False
+    """
+
+    values: List[Dict[str, Any]] = []
+
+    for sample_doc in samples:
+        values.append(map_mongo_sample_to_mysql(sample_doc))
+
+    mysql_conn = create_mysql_connection(config, False)
+
+    if mysql_conn is not None and mysql_conn.is_connected():
+        try:
+            run_mysql_executemany_query(mysql_conn, SQL_MLWH_MULTIPLE_INSERT, values)
```

Comment:
Add test to check on update SQL_MLWH_MULTIPLE_INSERT updats must_seq and pref_seq